frontend/src/routes/customer/escort/$escortId/$helperId.helper.$applicationId.tsx

```diff
@@ -4,18 +4,41 @@ import { KeywordTag, ReviewCard, SatisfactionGraph, StatsSummaryCard } from '@cu
 import { useModal } from '@hooks';
 import { IcPhoneFill, IcVerified } from '@icons';
 import { PageLayout } from '@layouts';
-import { createFileRoute } from '@tanstack/react-router';
+import { createFileRoute, useNavigate, useParams, useSearch } from '@tanstack/react-router';
 import type { EscortStrength } from '@types';
+import { dateFormat, formatImageUrl } from '@utils';
 
-export const Route = createFileRoute('/customer/escort/$escortId/helper/$helperId')({
+export const Route = createFileRoute('/customer/escort/$escortId/$helperId/helper/$applicationId')({
+  validateSearch: (search: { canSelect?: string } | undefined) => ({
+    canSelect: search?.canSelect,
+  }),
   component: RouteComponent,
 });
 
 function RouteComponent() {
+  const navigate = useNavigate();
   const { isOpen, openModal, closeModal } = useModal();
-  const { helperId } = Route.useParams();
+  const { helperId, escortId, applicationId } = useParams({
+    from: '/customer/escort/$escortId/$helperId/helper/$applicationId',
+  });
+  const { canSelect } = useSearch({
+    from: '/customer/escort/$escortId/$helperId/helper/$applicationId',
+  });
```

Comment:
```suggestion
  const { canSelect } = Route.useSearch();
```

현재 라우팅 컴포넌트니 요렇게만 해줘도 될 것 같습니다.

추가로 라우트 페이지가 아니라 내부 컴포넌트의 경우에는 getRouteApi라는 함수로 처리가 가능하다고 합니다.

[Search Params in Components](https://tanstack.com/router/latest/docs/framework/react/guide/search-params#search-params-in-components)

관련 내용 정리되어있는 부분이니 한 번 확인해보시면 좋을 것 같아요~